.github/workflows/syntax-checks.yml

```diff
@@ -1,6 +1,13 @@
 name: Syntax Checks
 
-on: [push]
+on:
```

Comment:
This is in conflict with https://github.com/nexusformat/definitions/pull/1024/files#diff-d679df740cd55f9498164266c52f7939b5b0fd15c83cece4a835a5473aee7640R3

In #1024, syntax checks will be run on branches that have a pull request to the repository.  This PR implements a change so that syntax checks are run only on the main branch.

Reply:
I want this immediate feedback on any push.

Reply:
For me, that is long-desired effect of a _continuous_ integration process.  The PR proposes an _occasional_ integration process.

Reply:
I think it is good practice to always have a pull request.

Reply:
If your branch has a PR you have as many PR events as push events.

Reply:
Indeed, it's just to avoid unnecessary CI usage. Basically you only want CI when you either push to the `main` branch or create/modify a pull request to the `main` branch.